tests/test_md.py

```diff
@@ -907,3 +915,99 @@ def build_context(barostat_interval):
+    nb_pot = summed_bound.potential.potentials[nb_pot_idx]
+    nb_bound = nb_pot.bind(summed_bound.potential.params_init[nb_pot_idx])
+    nb_pot_precomputed_idx = next(
+        i for i, pot in enumerate(summed_bound.potential.potentials) if isinstance(pot, NonbondedPairListPrecomputed)
+    )
+    nb_pot_precomputed = summed_bound.potential.potentials[nb_pot_precomputed_idx]
+    nb_bound_precomputed = nb_pot_precomputed.bind(summed_bound.potential.params_init[nb_pot_precomputed_idx])
+    # Construct nonbonded potentials from the reference one, will modify with atom indices to validate local MD produces
+    # the same results.
+    all_pairs = NonbondedAllPairs(
+        nb_pot.num_atoms,
+        nb_pot.beta,
+        nb_pot.cutoff,
+        disable_hilbert_sort=nb_pot.disable_hilbert_sort,
+        atom_idxs=non_ligand_idxs,
+        nblist_padding=nb_pot.nblist_padding,
+    ).bind(nb_bound.params)
+    ixn_group = NonbondedInteractionGroup(
+        nb_pot.num_atoms,
+        ligand_idxs,
+        nb_pot.beta,
+        nb_pot.cutoff,
+        disable_hilbert_sort=nb_pot.disable_hilbert_sort,
+        nblist_padding=nb_pot.nblist_padding,
+    ).bind(nb_bound.params)
+    exclusion_idxs, scale_factors = nonbonded.filter_exclusions(
+        non_ligand_idxs, nb_pot.exclusion_idxs, nb_pot.scale_factors, update_idxs=False
+    )
+    exclusions = NonbondedPairListNegated(exclusion_idxs, scale_factors, nb_pot.beta, nb_pot.cutoff).bind(
+        nb_bound.params
+    )
+
+    ref_bps = [fn.to_gpu(np.float32).bound_impl for fn in bound_pots]
+
+    # Construct without nonbonded potential and add modified all pairs, precomputed, ixn group and exclusions
+    comp_bps = [fn.to_gpu(np.float32).bound_impl for fn in bound_pots if not isinstance(fn.potential, SummedPotential)]
+
+    all_pairs_bound = all_pairs.to_gpu(np.float32).bound_impl
+
+    # Add back in the potentials that have been set up to have the ligand -> env computed with ixn group
+    comp_bps.append(nb_bound_precomputed.to_gpu(np.float32).bound_impl)
+    comp_bps.append(all_pairs_bound)
+    comp_bps.append(exclusions.to_gpu(np.float32).bound_impl)
+    comp_bps.append(ixn_group.to_gpu(np.float32).bound_impl)
+
+    v0 = np.zeros_like(coords)
+
+    intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+
+    steps = 100
+
+    ctxt = custom_ops.Context(coords, v0, box, intg.impl(), ref_bps)
+    ctxt.setup_local_md(temperature, True)
+    ref_xs, ref_boxes = ctxt.multiple_steps(steps)
+    ref_local_xs, ref_local_boxes = ctxt.multiple_steps_local(steps, ligand_idxs, burn_in=0)
+
+    ctxt = custom_ops.Context(coords, v0, box, intg.impl(), comp_bps)
+    ctxt.setup_local_md(temperature, True)
+    test_xs, test_boxes = ctxt.multiple_steps(steps)
+    test_local_xs, test_local_boxes = ctxt.multiple_steps_local(steps, ligand_idxs, burn_in=0)
+
+    # Verify that the all pairs is set back to the original atom indices set
```

Comment:
Not sure I understand what "set back" means here. It looks like `all_pairs` is initialized with `non_ligand_idxs` on L958. Are the indices expected to change?

Reply:
Oh, is this because `multiple_steps_local` modifies the atom indices?

Reply:
Never mind -- was confused at first but I think it makes sense.